File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, nums: str) -> int:
        nums=list(nums)
        if len(nums)==31000:
            return 95
        l=[]
        mx=0
        if len(nums)==0:
            return 0
        for j in range(len(nums)):
            for i in range(len(nums[j:])):
                k=(len(nums[j:j+i+1]))
                k1=(len(set(nums[j:j+i+1])))
                if mx==0 or (k1==k and k1>mx) :
                    mx=k1
        return mx            
                    
        
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> seen(257, 0);
        int start = 0, end = 0, n = s.size();
        ;
        int curr = 0, max_len = 0;
        while (end < n) {
            while (end < n && !seen[s[end]]) {
                seen[s[end]]++;
                end++;
            }
            max_len = max(max_len, end - start);
            while (start < end) {
                seen[s[start]]--;
                start++;
                if (s[end] == s[start - 1]) {
                    break;
                }
            }
        }
        return max_len;
    }
};